app/controllers/feedback_controller.rb

```diff
@@ -1,35 +1,49 @@
 class FeedbackController < ApplicationController
+  include OrganizationScopable
+
   skip_before_action :authenticate_user!
   skip_verify_authorized only: %i[new create]
-  layout :set_layout, only: %i[new create]
+  layout :set_layout, only: %i[new]
 
   def new
-    @contact = Contact.new
+    @feedback = Feedback.new
   end
 
   def create
-    @contact = Contact.new(contact_params)
-    if @contact.valid?
-      FeedbackMailer.with(contact_params).send_message.deliver_later
-      redirect_to root_path, notice: I18n.t("contacts.create.success")
+    @feedback = Feedback.new(feedback_params)
+
+    if @feedback.valid?
+      FeedbackMailer.with(feedback_params).send_message.deliver_now
+      redirect_to path, notice: I18n.t("contacts.create.success")
     else
       render :new, status: :unprocessable_entity
     end
   end
 
   private
 
-  def contact_params
-    params.require(:contact).permit(:name, :email, :message)
+  def feedback_params
+    params.require(:feedback).permit(:name, :email, :message, :subject)
   end
 
   def set_layout
     if current_user.nil?
       "application"
-    elsif current_user.staff_account
+    elsif current_user.has_role?(:adopter, ActsAsTenant.current_tenant)
+      "adopter_foster_dashboard"
+    else
       "dashboard"
+    end
+  end
+
+  def path
```

Comment:
I have been starting at this method as it just doesn't quite feel 'right'. We are defining the path based on the layout. It works. But the layout is defined based on whether there is no `current_user`, the user is an adopter, the user is a staff. It seems to me we should use the same checks for the redirect path. Using the layout feels like it's a proxy, and would break if we ever changed the layouts. The driver here should be the user type, or lack thereof. Thoughts on this?

It's not a hill I am going to die on...but I think it would be clearer.

Reply:
For sure. I think I originally was trying to keep things DRY and use the set_layout method and I kinda just went down that hole. Then I was lazy and just moved the case into a method without thinking about it more.

Maybe something like
```
  def path
    if current_user.nil?
     root_path
    elsif current_user.has_role?(:adopter, ActsAsTenant.current_tenant)
      adopter_fosterer_dashboard_index_path
    else
      staff_dashboard_index_path
    end
  end
```

Same as set_layout but returning the path instead. Maybe there is a simple way to use set_layout for both but I can't think of it at the moment or even if that is a smart idea.

Reply:
Yes I think that works ^. I also wondered about setting both at once...but single responsibility is probably better to stick to when in doubt!